packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -16,11 +17,24 @@ import {
     IProvideFluidDataStoreRegistry,
 } from "@fluidframework/runtime-definitions";
 
-interface IResponseException extends Error {
-    errorFromRequestFluidObject: true;
-    message: string;
-    code: number;
-    stack: string;
+class ResponseException extends LoggingError {
+    public errorFromRequestFluidObject: true = true;
+    public code: number;
+    constructor(message: string, response: IResponse) {
+        super(message);
+        this.code = response.status;
+        // Tag response.value as PackageData because it typically contains the request URL
```

Comment:
That's not true. First of all, it will not contain PackageData, it will contains data store ID, potentially DDS id, which might still contain name that 3rd party developer assigned to it, so it should be properly categorized as "code artifact" (if name comes from code). I think package name tag should be renamed to be "code artifact" to be more inclusive and include cases like that (stack falls into same bucket).
But it could contain user data (query param for sure, even data store IDs might theoretically be produced from user data, though very unlikely), so using contamination model, whole payload should be annotated as containing user data.

That's the reason we do not actually collect URIs right now - see this PR:
https://github.com/microsoft/FluidFramework/pull/5327/files

So I think we have two options:
1. Do nothing - we do not have any violations
2. (Ideal, to help with debuggability) Start collecting only base URI (not query params) and annotate it as code artifact.

We likely also should put a warning somewhere (on create APIs for data stores & DDSs) that ID provided will be collected in telemetry using this tag. That said, I think it's relatively safe either way, as we are pushing developers to give names only to root data stores / DDSs, so there is very little risk user input would be used here.

Reply:
Ok great info, thanks!  I like the term CodeArtifact as well, I used that name in an earlier iteration before switching to PackageInfo so happy to revert back :)

Reply:
I'm going to rename to CodeArtifact in a follow-up PR